tests/dedicatedevhtests/l7_evh_crd_test.go

```diff
@@ -0,0 +1,440 @@
+
+	integrationtest.AddConfigMap(KubeClient)
+	ctrl.HandleConfigMap(informers, ctrlCh, stopCh, quickSyncCh)
+	integrationtest.KubeClient = KubeClient
+	integrationtest.AddDefaultIngressClass()
+	keyChan = make(chan string)
+	ctrl.SetSEGroupCloudNameFromNSAnnotations()
+	go ctrl.InitController(informers, registeredInformers, ctrlCh, stopCh, quickSyncCh, waitGroupMap)
+	os.Exit(m.Run())
+}
+
+func SetupDomain() {
+	mcache := cache.SharedAviObjCache()
+	cloudObj := &cache.AviCloudPropertyCache{Name: "Default-Cloud", VType: "mock"}
+	subdomains := []string{"avi.internal", ".com"}
+	cloudObj.NSIpamDNS = subdomains
+	mcache.CloudKeyCache.AviCacheAdd("Default-Cloud", cloudObj)
+}
+
+func SetUpTestForIngress(t *testing.T, modelNames ...string) {
+	for _, model := range modelNames {
+		objects.SharedAviGraphLister().Delete(model)
+	}
+	integrationtest.CreateSVC(t, "default", "avisvc", corev1.ServiceTypeClusterIP, false)
+	integrationtest.CreateEP(t, "default", "avisvc", false, false, "1.1.1")
+}
+
+func TearDownTestForIngress(t *testing.T, modelNames ...string) {
+	// for _, model := range modelNames {
+	// 	objects.SharedAviGraphLister().Delete(model)
+	// }
+	integrationtest.DelSVC(t, "default", "avisvc")
+	integrationtest.DelEP(t, "default", "avisvc")
+}
+
+func SetUpIngressForCacheSyncCheck(t *testing.T, tlsIngress, withSecret bool, modelNames ...string) {
+	SetupDomain()
+	SetUpTestForIngress(t, modelNames...)
+	ingressObject := integrationtest.FakeIngress{
+		Name:        "foo-with-targets",
+		Namespace:   "default",
+		DnsNames:    []string{"foo.com"},
+		Ips:         []string{"8.8.8.8"},
+		HostNames:   []string{"v1"},
+		Paths:       []string{"/foo"},
+		ServiceName: "avisvc",
+	}
+	if withSecret {
+		integrationtest.AddSecret("my-secret", "default", "tlsCert", "tlsKey")
+	}
+	if tlsIngress {
+		ingressObject.TlsSecretDNS = map[string][]string{
+			"my-secret": {"foo.com"},
+		}
+	}
+	ingrFake := ingressObject.Ingress()
+	if _, err := KubeClient.NetworkingV1().Ingresses("default").Create(context.TODO(), ingrFake, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("error in adding Ingress: %v", err)
+	}
+	integrationtest.PollForCompletion(t, modelNames[0], 5)
+}
+
+func TearDownIngressForCacheSyncCheck(t *testing.T, modelName string) {
+	if err := KubeClient.NetworkingV1().Ingresses("default").Delete(context.TODO(), "foo-with-targets", metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("Couldn't DELETE the Ingress %v", err)
+	}
+	KubeClient.CoreV1().Secrets("default").Delete(context.TODO(), "my-secret", metav1.DeleteOptions{})
+	TearDownTestForIngress(t, modelName)
+}
+
+func TestCreateUpdateDeleteSSORuleForEvh(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	modelName, _ := GetDedicatedModel("foo.com", "default")
+	srname := "samplesr-foo"
+	SetUpIngressForCacheSyncCheck(t, true, true, modelName)
+
+	err := integrationtest.SetUpOAuthSecret()
+	if err != nil {
+		t.Fatalf("error in creating my-oauth-secret: %v", err)
+	} else {
```

Comment:
`else` is not required here since `t.Fatalf` will exit the program.

Reply:
removed